### PR TITLE
feat: [AXIMST-694] make sidebar fixed on scroll and add tests

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**Description:** 
- make course outline sidebar fixed on scroll and depends on content height
- add test to course-outline sidebar components

**Youtrack:** [AXIMST-694](https://youtrack.raccoongang.com/issue/AXIMST-694/Adjust-Tests-coverage)

### CODE REVIEW

File: src/courseware/course/sidebar/sidebars/course-outline/SidebarUnit.jsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ const SidebarUnit = ({
   const {
     complete,
     title,
-    icon,
+    icon = UNIT_ICON_TYPES.other,
   } = unit;
```
Comment (ihor-romaniuk) [inline src/courseware/course/sidebar/sidebars/course-outline/SidebarUnit.jsx:23]: Add coverage for `const iconType = isLocked ? UNIT_ICON_TYPES.lock : icon;` after merge https://github.com/raccoongang/frontend-app-learning/pull/24

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.test.jsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  it('renders correctly when course outline is loading', async () => {
+    await initTestStore({ excludeFetchOutlineSidebar: true });
+    renderWithProvider();
+
+    expect(screen.getByText(messages.loading.defaultMessage)).toBeInTheDocument();
+    expect(screen.getByText(messages.courseOutlineTitle.defaultMessage)).toBeInTheDocument();
+    expect(screen.queryByRole('button', { name: 'Course Outline' })).not.toBeInTheDocument();
+  });
+
+  it('doesn\'t renders when outline sidebar is disabled', async () => {
```
Comment (monteri) [inline src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.test.jsx]:
```suggestion
  it('doesn\'t render when outline sidebar is disabled', async () => {
```

File: src/courseware/course/sidebar/sidebars/course-outline/SidebarSection.test.jsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  it('renders correctly when section is incomplete', () => {
+    const { getByText, container } = render(<RootWrapper />);
+
+    expect(getByText(section.title)).toBeInTheDocument();
+    expect(screen.getByText(`, ${courseOutlineMessages.incompleteSection.defaultMessage}`)).toBeInTheDocument();
+    expect(container.querySelector('.text-success')).not.toBeInTheDocument();
+
+    const button = getByText(section.title);
+    fireEvent.click(button);
```
Comment (monteri) [inline src/courseware/course/sidebar/sidebars/course-outline/SidebarSection.test.jsx]:
```suggestion
    userEvent.click(button);
```

File: src/courseware/course/sidebar/sidebars/course-outline/SidebarSection.test.jsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    const { getByText, container } = render(
+      <RootWrapper section={{ ...section, complete: true }} />,
+    );
+
+    expect(getByText(section.title)).toBeInTheDocument();
+    expect(screen.getByText(`, ${courseOutlineMessages.completedSection.defaultMessage}`)).toBeInTheDocument();
+    expect(container.querySelector('.text-success')).toBeInTheDocument();
+
+    const button = getByText(section.title);
+    fireEvent.click(button);
```
Comment (monteri) [inline src/courseware/course/sidebar/sidebars/course-outline/SidebarSection.test.jsx]:
```suggestion
    userEvent.click(button);
```

File: src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx [hunk 5]
```diff
@@ -40,7 +40,11 @@ const UnitIcon = ({ type, isCompleted, ...props }) => {
     },
   };
 
-  const Icon = iconMap[type]?.[isCompleted ? 'complete' : 'default'];
+  let Icon = iconMap[type || UNIT_ICON_TYPES.other];
+
+  if (typeof iconMap[type || UNIT_ICON_TYPES.other] === 'object') {
+    Icon = iconMap[type || UNIT_ICON_TYPES.other]?.[isCompleted ? 'complete' : 'default'];
+  }
```
Comment (monteri) [inline src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx]:
```suggestion
const defaultIconType = type || UNIT_ICON_TYPES.other;
let Icon = iconMap[defaultIconType];

if (typeof Icon === 'object') {
  const iconName = isCompleted ? 'complete' : 'default';
  Icon = Icon[iconName];
}
```

**Review by monteri — CHANGES_REQUESTED** [review]

**Comment by monteri** [comment]
Also rebase required?

File: src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.test.jsx (hunk 2 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.test.jsx]: Changed

File: src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx (hunk 5 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx]: Changed in previous PR

File: src/courseware/course/sidebar/sidebars/course-outline/SidebarSection.test.jsx (hunk 4 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/SidebarSection.test.jsx]: Changed

File: src/courseware/course/sidebar/sidebars/course-outline/SidebarSection.test.jsx (hunk 3 above)
Comment (ihor-romaniuk) [reply src/courseware/course/sidebar/sidebars/course-outline/SidebarSection.test.jsx]: Changed

**Pull request merged**